src/main/java/org/keycloak/plugin/rhmi/DelayAuthentication.java

```diff
@@ -0,0 +1,165 @@
+import java.util.List;
+import java.util.Locale;
+
+import javax.ws.rs.core.Response;
+import org.keycloak.utils.MediaType;
+
+public class DelayAuthentication extends AbstractIdpAuthenticator implements Authenticator {
+
+	private static final String USER_CREATED_ATTRIBUTE = "3scale_user_created";
+	private static final String USER_CREATED_VALUE = "true";
+	private static final Logger logger = Logger.getLogger(DelayAuthentication.class);
+
+	
+	public void close() {
+	}
+	
+	/**
+	 * checks whether user has been created in keycloak and received creation attribute
+	 * and received the created attribute set to true 
+	 * @param user
+	 * @return
+	 */
+	private boolean isUserCreated(UserModel user) {
+
+		if (user == null) {
+			logger.debug("User is null");
+			return false;
+		}
+
+		String userCreatedAtt = user.getFirstAttribute(DelayAuthentication.USER_CREATED_ATTRIBUTE);
+		if (userCreatedAtt == null) {
+			logger.debugf("User is created but %s attribute in keycloak is null", DelayAuthentication.USER_CREATED_ATTRIBUTE);
+			return false;
+		}
+		
+		if (!userCreatedAtt.equals(DelayAuthentication.USER_CREATED_VALUE)) {
+			logger.debugf("%s attribute value in keycloak is not equals to %s", 
+					DelayAuthentication.USER_CREATED_ATTRIBUTE, 
+					DelayAuthentication.USER_CREATED_VALUE
+			);
+			return false;
+		}
+		
+		return true;
+	}
+	
+
+
+	public boolean requiresUser() {
+		return false;
+	}
+
+	public boolean configuredFor(KeycloakSession session, RealmModel realm, UserModel user) {
+		return false;
+	}
+
+	public void setRequiredActions(KeycloakSession session, RealmModel realm, UserModel user) {
+	}
+	
+
+	@Override
+	protected void authenticateImpl(AuthenticationFlowContext context, SerializedBrokeredIdentityContext serializedCtx,
+			BrokeredIdentityContext brokerContext) {
+
+		if (isUserCreated(context.getUser())) {
+			redirectToAfterFirstBrokerLoginSuccess(context, brokerContext);
+		} else {
+			redirectToAction(context);
+		}		
+	}
+
+	@Override
+	protected void actionImpl(AuthenticationFlowContext context, SerializedBrokeredIdentityContext serializedCtx,
+			BrokeredIdentityContext brokerContext) {
+		
+        // get user by federated identity - IdentityBrokerService method authenticated 
+        KeycloakSession session = context.getSession();
+        FederatedIdentityModel federatedIdentityModel = new FederatedIdentityModel(brokerContext.getIdpConfig().getAlias(), brokerContext.getId(),
+        		brokerContext.getUsername(), brokerContext.getToken());
+ 
+        RealmModel realm = context.getRealm();
+        
+        UserModel federatedUser = session.users().getUserByFederatedIdentity(federatedIdentityModel, realm);
+        context.setUser(federatedUser);
+
+        if (isUserCreated(context.getUser())) {
+        	redirectToAfterFirstBrokerLoginSuccess(context, brokerContext);
+		} else {
+			redirectToAction(context);
+		}
+	}
+	
+	private void redirectToAction(AuthenticationFlowContext context) {
```

Comment:
I think this name isn't very precise. Maybe something like `showAccountProvisioningPage` ?